helm/prometheus-agent/charts/prometheus-agent/values.yaml

```diff
@@ -8,6 +8,10 @@ image:
   sha: ""
   pullPolicy: "Always"
 
+pdb:
+  minAvailable: 0
```

Comment:
Are you sure we want to allow 0 pod ?
I thought we allow to reschedule the pod on another node, so `minAvailable: 1`

Reply:
If we set minAvailable: 1 with only one pod, it will never be evicted right?

Reply:
How I understand pdb is:
`minAvailable: 1` means if the node where the agent is running has to be drained, the scheduler has to reschedule the pod on another node.
It prevents to scale down to 0 our prometheus-agent.

Reply:
with 0 it will block rolling of nodes when we do a cluster upgrade because the node running that pod will not be able to get the pod evicted 

does it even make sense to have a PDB for single pod workloads ?

Reply:
The way I see it, minAvailable: 1 means there should always be 1 pod running but I;ll test :)

Reply:
@primeroz even with a maxUnavailable of 100%? This is confusing. It does make sense because this is blocking the cluster autoscaler from evicting nodes as it's running in kube-system. Maybe the best way is to do it the same way we did it for metrics-server-app

Reply:
when a node is drained an `evict` event of all pods running on the node is issued, evict will always `stop the pod` and recreate it somewhere else so with a `min: 1` on a workload with `1 replica` you block that node from rolling 

see https://github.com/giantswarm/vertical-pod-autoscaler-app/pull/177

Reply:
Hummm I'm not sure pdb is the solution:
https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues/1934

I try to find something relevant.

Reply:
Sure but this is causing issues : https://github.com/giantswarm/giantswarm/issues/26883

Reply:
> So why would minAvailable: 0 block eviction if 1 does?

oh no sorry, `minAvailable: 0` is fine ... but then again, what's the point of such a PDB ?

Reply:
> Sure but this is causing issues : https://github.com/giantswarm/giantswarm/issues/26883

weird , i never seen that issue before with cluster-autoscaler 

yeah , i guess that explain the `pdb with minAvailable:0` . just not `1` :+1:

Reply:
have you tried to add the `safe-to-evict` annotation ? https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#what-types-of-pods-can-prevent-ca-from-removing-a-node

Reply:
Also, have we considered not running `prometheus-agent` in `kube-system` ? 

pods in `kube-system` are handled differently by CA and they do require a PDB  , while single replica pods in any other namespace would be evicted no problem

Reply:
I did not think about this one "cluster-autoscaler.kubernetes.io/safe-to-evict": "true" That would probably be best :)

Reply:
Let's test with minAvailability: 1 and be sure the agent will be rescheduled on another pod after node deletion.

Reply:
:+1: `minAvailability: 1` will definetely break node rolling